src/jit/lower.cpp

```diff
@@ -3408,6 +3408,9 @@ bool Lowering::IndirsAreEquivalent(GenTreePtr candidate, GenTreePtr storeInd)
     if (pTreeA->OperGet() != pTreeB->OperGet())
         return false;
 
+    if (candidate->gtType != storeInd->gtType)
```

Comment:
Anyways, I like your fix, though I'd make the genActualType check an assert only.  I talked to a few others over here, and it should be impossible to get differing types like int/float at this point without an explicit cast node, based on the IR design.